scene/gui/text_edit.cpp

```diff
@@ -2927,6 +2951,35 @@ void TextEdit::_update_ime_text() {
 	queue_redraw();
 }
 
+void TextEdit::_on_focus_enter() {
+	if (caret_blink_enabled) {
+		caret_blink_timer->start();
+	} else {
+		draw_caret = true;
+	}
+
+	_update_ime_window_position();
+
+	if (DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_VIRTUAL_KEYBOARD) && virtual_keyboard_enabled) {
```

Comment:
```suggestion
	if (virtual_keyboard_enabled && DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_VIRTUAL_KEYBOARD)) {
```
Faster check first

Reply:
its not in the hotpath and wasnt causing problems, but ill make the change